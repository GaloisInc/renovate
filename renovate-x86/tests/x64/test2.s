	.file	"test2.c"
	.globl	g
	.data
	.align 4
	.type	g, @object
	.size	g, 4
g:
	.long	-11
	.text
	.globl	_start
	.type	_start, @function
_start:
.LFB0:
	.cfi_startproc
	pushq	%rbp
	.cfi_def_cfa_offset 16
	.cfi_offset 6, -16
	movq	%rsp, %rbp
	.cfi_def_cfa_register 6
	movl	g(%rip), %eax
	testl	%eax, %eax
	jle	.L2
	movl	g(%rip), %eax
	addl	$1, %eax
	movl	%eax, g(%rip)
.L2:
#APP
# 10 "test2.c" 1
	movq $60,%rax
movq $0,%rdi
syscall
# 0 "" 2
#NO_APP
	nop
	popq	%rbp
	.cfi_def_cfa 7, 8
	ret
	.cfi_endproc
.LFE0:
	.size	_start, .-_start
	.ident	"GCC: (Ubuntu 6.2.0-5ubuntu12) 6.2.0 20161005"
	.section	.note.GNU-stack,"",@progbits
