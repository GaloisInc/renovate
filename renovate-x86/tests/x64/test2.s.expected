ExpectedResult { expectedBlocks = [
  ExpectedBlock { addr = 0x2b1
                , byteCount = 14
                , insnCount = 5
                },
  ExpectedBlock { addr = 0x2bf
                , byteCount = 15
                , insnCount = 3
                },
  ExpectedBlock { addr = 0x2ce
                , byteCount = 16
                , insnCount = 3
                }
  ],
  ignoreBlocks = [0x2de]
}